internal/provider/resource_gitlab_group_saml_link.go

```diff
@@ -0,0 +1,125 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_group_saml_link", func() *schema.Resource {
+	validGroupSamlLinkAccessLevelNames := []string{
+		"Guest",
+		"Reporter",
+		"Developer",
+		"Maintainer",
+		"Owner",
```

Comment:
We usually write down all lower-case and they only work that way with the `accessLevelNameToValue` helper:

```suggestion
		"guest",
		"reporter",
		"developer",
		"maintainer",
		"owner",
```